src/clj/puppetlabs/http/client/async.clj

```diff
@@ -175,12 +176,13 @@
   ([opts :- common/RawUserRequestOptions
     callback :- common/ResponseCallbackFn
     client :- HttpAsyncClient]
-    (request-with-client opts callback client nil nil))
+    (request-with-client opts callback client nil nil true))
   ([opts :- common/RawUserRequestOptions
     callback :- common/ResponseCallbackFn
     client :- HttpAsyncClient
     metric-registry :- (schema/maybe MetricRegistry)
-    metric-namespace :- (schema/maybe schema/Str)]
+    metric-namespace :- (schema/maybe schema/Str)
+    use-url-metrics :- (schema/maybe schema/Bool)]
```

Comment:
`request-with-client` is only sort-of public, I guess, though.  Seems like most clients would get to this via a call to one of the wrappers in the `sync` or `async` namespace.  So I guess I'd be okay with doing it the way you have it.  I think @rlinehan's suggestion to make the new parameter required makes sense so we can avoid having to deal with a `nil` value.